dbuf-lsp/src/core/navigator/indentify.rs

```diff
@@ -18,123 +18,128 @@ struct GetImpl<'a> {
-        };
+        }
     }
 
-    fn return_field(&mut self, field: &Str) {
-        assert!(self.no_result());
-        self.result = Symbol::Field {
+    fn get_field(&self, field: &Str) -> Symbol {
+        assert!(field.get_location().contains(self.target));
+
+        Symbol::Field {
+            t: self.scope.get_type().to_owned(),
             constructor: self.scope.get_constructor().to_owned(),
             field: field.to_string(),
-        };
+        }
     }
 
-    fn return_alias(&mut self, alias: &Str) {
-        assert!(self.no_result());
-        self.result = Symbol::Alias {
+    fn get_alias(&self, alias: &Str) -> Symbol {
+        assert!(alias.get_location().contains(self.target));
+
+        Symbol::Alias {
             t: self.scope.get_type().to_owned(),
             branch_id: self.scope.get_branch_id(),
-            name: alias.to_string(),
-        };
+            alias: alias.to_string(),
+        }
     }
 
-    fn return_constructor(&mut self, constructor: &Str) {
-        assert!(self.no_result());
+    fn get_constructor(&self, constructor: &Str) -> Symbol {
+        assert!(constructor.get_location().contains(self.target));
 
         if self.elaborated.is_message(constructor.as_ref()) {
-            self.result = Symbol::Type(constructor.to_string());
+            Symbol::Type(constructor.to_string())
         } else {
-            self.result = Symbol::Constructor(constructor.to_string());
+            Symbol::Constructor {
+                t: self.scope.get_type().to_owned(),
+                constructor: constructor.to_string(),
+            }
         }
     }
 
-    fn return_access(&mut self, access: &Str) {
+    fn get_access(&self, access: &Str) -> Symbol {
         assert!(access.get_location().contains(self.target));
 
-        // Variable should be either dependency or field
+        // Variable should be one of: dependency, field, alias
         if self
             .elaborated
             .is_type_dependency(self.scope.get_type(), access.as_ref())
         {
-            self.return_dependency(access);
+            self.get_dependency(access)
         } else if self
             .elaborated
             .is_constructor_field(self.scope.get_constructor(), access.as_ref())
         {
-            self.return_field(access);
+            self.get_field(access)
+        } else if self
+            .elaborated
+            .is_constructor_implicit(self.scope.get_constructor(), access.as_ref())
+        {
+            self.get_alias(access)
         } else {
             panic!("bad variable expr")
         }
     }
 }
 
 impl<'a> Visitor<'a> for GetImpl<'a> {
-    fn visit(&mut self, visit: Visit<'a>) -> VisitResult {
+    type StopResult = Symbol;
+
+    fn visit(&mut self, visit: Visit<'a>) -> VisitResult<Self::StopResult> {
         match &visit {
             Visit::Keyword(_, _) => {}
             Visit::Type(type_name, type_location) => {
                 if !type_location.contains(self.target) {
```

Comment:
You can use [match guards](https://doc.rust-lang.org/book/ch19-03-pattern-syntax.html#extra-conditionals-with-match-guards) to make pattern-matching more declarative and use value-return instead of explicit `return` commands